src/io_threads.c

```diff
@@ -439,13 +439,29 @@ void IOThreadFreeArgv(void *data) {
 }
 
 /* This function attempts to offload the client's argv to an IO thread.
+ *
+ * If free_original_argv is set to 1, the function will offload the free of the client's original argv.
+ * Otherwise, it will offload the free of the client's current argv.
+ *
  * Returns C_OK if the client's argv were successfully offloaded to an IO thread,
  * C_ERR otherwise. */
-int tryOffloadFreeArgvToIOThreads(client *c) {
+int tryOffloadFreeArgvToIOThreads(client *c, int free_original_argv) {
```

Comment:
Here, it is clear what the arguments mean, but at the call site, it is not immediately clear to the reader what the 2nd argument is, when you see the call like `tryOffloadFreeArgvToIOThreads(c, 0)`.

Trying to think of ideas:

* We could use some `enum {FREE_ARGV, FREE_ORIGINAL_ARGV}` instead of 0 and 1.
* We could pass pointers to the argv and argc fields within the client struct instead of the current parameters, so `tryOffloadFreeArgvToIOThreads(robj ***argv_ptr, int *argc_ptr)` and then at the call site it would look like `tryOffloadFreeArgvToIOThreads(&c->original_argv, &c->original_argc)`. Is this good?

Reply:
Thanks, good suggestion, changed to `tryOffloadFreeArgvToIOThreads(client *c, int argc, robj **argv)` no need for `***argv_ptr`

Reply:
Good idea. That is even better.